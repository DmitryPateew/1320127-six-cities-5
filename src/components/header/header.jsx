import React from "react";
import {AuthorizationStatus, PagePath} from "../../constant";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import PropTypes from "prop-types";

const Header = (props) => {
  const {authorizationStatus, user} = props;
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to={PagePath.MAIN} className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <span className="header__nav-link header__nav-link--profile">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                  </div>
                  {authorizationStatus === AuthorizationStatus.AUTH ?
                    <Link className="header__user-name user__name" to={PagePath.FAVORITE}>
                      {user ? user.email : ``}
                    </Link>
                    : <Link className="header__login" to={PagePath.LOGIN}>Sign in</Link>}
                </span>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
  user: USER.user,
});

Header.propTypes = {
  authorizationStatus: PropTypes.string,
  login: PropTypes.string,
  user: PropTypes.shape({
    email: PropTypes.string,
  })
};
export {Header};
export default connect(mapStateToProps)(Header);
