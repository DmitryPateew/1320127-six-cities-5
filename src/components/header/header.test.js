import React from "react";
import {Header} from "./header";
import {expect, it} from "@jest/globals";
import Enzyme, {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

const userTest = {email: `valera32@mail.ru`};

it(`Should Header render correctly`, () => {
  const header = shallow(<Header
    authorizationStatus={`AUTH`}
    user={userTest}
  />);

  expect(shallowToJson(header)).toMatchSnapshot();
});
