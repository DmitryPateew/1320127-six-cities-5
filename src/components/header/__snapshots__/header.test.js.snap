// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <Link
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </Link>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <span
              className="header__nav-link header__nav-link--profile"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <Link
                className="header__user-name user__name"
                to="/favorites"
              >
                valera32@mail.ru
              </Link>
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
